content/src/debug.js

```diff
@@ -303,15 +623,44 @@ function parseTurtleTransform(transform) {
 }
 
 // Highlights the given line number as a line being traced.
-function traceLine(line) {
-  view.markPaneEditorLine(
-      view.paneid('left'), line, 'guttermouseable', true);
-  view.markPaneEditorLine(view.paneid('left'), line, 'debugtrace');
+function traceLine(lineIndex) {
+    var line = traceEvents[lineIndex].location.first_line;
+    var prevLine = -1;
+    var block_mode = true;
+
+    if (!view.getPaneEditorBlockMode(view.paneid("left"))){block_mode = false;}
+   
+    if (traceEvents[lineIndex-1]){
+      prevLine = traceEvents[lineIndex-1].location.first_line;
+    }
+    if (debugMode) { 
+      $('debugtraceprev').removeClass('inactive').addClass('active');
+      view.markPaneEditorLine(view.paneid('left'), line, 'debugtrace');
+      if (!block_mode) {
+        view.markPaneEditorLine(view.paneid('left'), prevLine, 'debugtraceprev');
+      }
+    }
+    view.markPaneEditorLine(
+    view.paneid('left'), line, 'guttermouseable', true);
+
+   
 }
 
 // Unhighlights the given line number as a line no longer being traced.
-function untraceLine(line) {
+function untraceLine(lineIndex) {
+  var line = traceEvents[lineIndex].location.first_line;
+  var prevLine = -1;
+  var block_mode = true;
```

Comment:
I don't understand this comment. What do you want me to write? Grabbing the block editor either returns the editor (in which case if (block_mode) is true) or returns nothing (then it's false). What's hard to understand about it?